### PR TITLE
Refactor the ResultType

### PR DESCRIPTION
**Pull request by mociarain**
### Which issue this PR addresses:

This PR is a small refactor that comes out of a pairing session with myself and @sankur-codes. The originating PR is [here](https://github.com/Azure/ARO-RP/pull/3396).

### Test plan for issue:

I've added some unit tests.

### Is there any documentation that needs to be updated for this PR?

Nope

### CODE REVIEW

**Review by tiguelu — APPROVED** [review]

**Comment by tiguelu** [comment]
Waiting for the auto-triggered E2E pipeline to complete before merging: https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=89282519&view=results 

For some reason, it's not linked within the checks above.

**Comment by tiguelu** [comment]
> Waiting for the auto-triggered E2E pipeline to complete: https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=89282519&view=results
> 
> For some reason, it's not auto-linked within the checks in the PR.

E2E succeeded.

File: pkg/frontend/frontend.go [hunk 1]
```diff
@@ -474,27 +474,23 @@ func frontendOperationResultLog(log *logrus.Entry, method string, err error) {
...
 	default:
-		log = log.WithField("resultType", utillog.ServerErrorResultType)
+		statusCode = 500
+	}
+
+	resultType := utillog.MapStatusCodeToResultType(statusCode)
+	log = log.WithField("resultType", resultType)
+
+	if resultType == utillog.SuccessResultType {
+		log.Info("long running operation succeeded")
```
Comment (kimorris27) [inline pkg/frontend/frontend.go:492]: I think this message was originally `front end operation succeeded` and should be reverted back to what it was.

**Review by kimorris27 — CHANGES_REQUESTED** [review]
The refactor makes sense! My requested change is to fix a log message that I think you probably changed unintentionally.

**Review by kimorris27 — APPROVED** [review]

**Review by tiguelu — APPROVED** [review]

**Pull request merged**